### ISSUE TITLE
publisher failing for index node without commercial SSL certificate

### ISSUE THREAD

**Issue by alaniwi** [ESGF/esg-publisher#83]
(I noticed the following with `esgunpublish`, and haven't verified whether `esgpublish` is similarly affected, though I guess it would be.)

When talking to a (test) index node that has an SSL certificate signed by ESGF but not commercially (in fact localhost, although using the full hostname in hessian_service_url), `esgunpublish` is failing with:

```
Deletion/retraction failed for dataset/version <dataset_id> with message: ESGPublishError("Socket error: SSLError(1, u'[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] ss:661)')\nIs the proxy certificate /root/.globus/certificate-file valid?",)
```

I found some suggestion that the default behaviour of httplib has changed to do SSL verification.

I've tested that this can be worked around by modifying `publish/hessianlib.py` to change:
```
ctx = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
```
to:
```
ctx = ssl._create_unverified_context()
```

It might be worth adding a command-line option to enable this behaviour - while perhaps leaving SSL verification as the default.

**Comment by LucaCinquini** [comment]
Potentially one could also experiment with using SSL_CERT_DIR and SSL_CERT_FILE to point to the directory/bundle that contain the ESGF CA…. if it works…
thanks, L

**Comment by alaniwi** [comment]
Good point. I haven't experimented with this, but it would be worth a try.

**Comment by alaniwi** [comment]
Obviously we'd need to be careful not to break publishing to commercially signed index nodes.

**Comment by sashakames** [comment]
So, this is only with self-signed certs?  All production index nodes should use commercially signed nodes, so a change would really only be needed with test deployments without them.

**Comment by alaniwi** [comment]
The problem was encountered on a test node. Agreed that production nodes should have commercial certs, but we don't want to require them on test nodes.

**Comment by sashakames** [comment]
I'm inclined to close this, as we can use letsencrypt for test nodes now.  I realize that in test cases, if the index node can't recognize the CA of the test data node, it may fail to harvest the catalog (haven't tested lately).
